File: ruby/problem_3.rb
#Problem: What is the largest prime factor of the number 600851475143 ?

number = 600851475143
upper_bound = Math.sqrt(number)
prime_factors = []

for i in 2..upper_bound
	if number % i == 0
		is_prime = true  #set to true by default until we find out otherwise
		prime_factors.each do |prime|
			if i % prime == 0 
				is_prime = false
				break
			end
		end
		prime_factors << i if is_prime
	end
end

puts prime_factors.last

File: python/problem_3.py
#Problem: What is the largest prime factor of the number 600851475143 ?


n = 600851475143
upper_bound = int(n**0.5) #squre root
prime_factors = []

for i in range(2,upper_bound):
    if n % i == 0:
        is_prime = True
        for prime in prime_factors:
            if i % prime == 0:
                prime = False
                break
        if is_prime:
            prime_factors.append(i)

print prime_factors[-1]
